java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -811,6 +972,12 @@ private void readFromJsonIntoVector(Field field, FieldVector vector)
         readToken(END_ARRAY);
       }
     }
+    if (variadicBufferIndices.isEmpty() && vector instanceof BaseVariableWidthViewVector) {
```

Comment:
Why can't the buffer reader take care of this?

Reply:
Removed.